File: android/app/src/main/kotlin/com/example/figure_skating_jumps/channels/enums/EventChannelParameters.kt
package com.example.figure_skating_jumps.channels.enums

import com.example.figure_skating_jumps.channels.event_channels.*
import io.flutter.plugin.common.EventChannel.StreamHandler

/**
 * A enum class that contains the event channel parameters
 *
 * @param channelName The name to identify the event channel
 * @param streamHandler The event channel stream handler
 */
enum class EventChannelParameters(val channelName: String, val streamHandler: StreamHandler) {
    BluetoothChannel("xsens-dot-bluetooth-permission", BluetoothPermissionStreamHandler),
    ConnectionChannel("xsens-dot-connection", XSensDotConnectionStreamHandler),
    MeasuringChannel("xsens-dot-measuring", XSensDotMeasuringStreamHandler),
    MeasuringStatusChannel("xsens-dot-measuring-status", XSensDotMeasuringStatusStreamHandler),
    RecordingChannel("xsens-dot-recording", XSensDotRecordingStreamHandler),
    ScanChannel("xsens-dot-scan", XSensDotScanStreamHandler)
}

File: ios/Plugins/Channels/Enums/EventChannelParameters.m
#import "EventChannelParameters.h"

NSString *ChannelEventNames(EventChannelParameters eventChannelParameters) {
    switch (eventChannelParameters) {
        case BluetoothChannelEvent:
            return @"xsens-dot-bluetooth-permission";
        case ConnectionChannelEvent:
            return @"xsens-dot-connection";
        case MeasuringChannelEvent:
            return @"xsens-dot-measuring";
        case MeasuringStatusChannelEvent:
            return @"xsens-dot-measuring-status";
        case RecordingChannelEvent:
            return @"xsens-dot-recording";
        case ScanChannelEvent:
            return @"xsens-dot-scan";
        default:
            return @"";
    }
}
